waspc/tests/Generator/Valid/ValidatorTest.hs

```diff
@@ -0,0 +1,284 @@
+module Generator.Valid.ValidatorTest (spec_Validator) where
+
+import Test.Hspec
+import qualified Wasp.Generator.Valid.Validator as V
+
+spec_Validator :: Spec
+spec_Validator = do
+  describe "execValidator" $ do
+    it "returns empty list when validation succeeds" $ do
```

Comment:
Here testSuccess is probably hiding too much because you are checking for empty list but what testSuccess check inside, who knows? So i would not use testSuccess here but write more direct code.